### PR TITLE
 added "Fail build if all/any scenarios failed" as an option

### PR DESCRIPTION
**Pull request by tommywo**
this is my fix for 
https://issues.jenkins-ci.org/browse/JENKINS-30081
screen:
screen shot 2015-11-17 at 15 49 05

### CODE REVIEW

**Comment by jtnord** [comment]
I'm not going to accept this on principal that it has a travis file...

**Comment by jtnord** [comment]
Thank you for you contribution.
Fail if any and fail if all should not be 2 checkboxes as all implies any. So this should be a failed test dropdown which sets a behaviour (defaulting to mark unstable if any failures)
It should also be in an advanced section as I think most users would not want to change this.

**Comment by tommywo** [comment]
I've changed two checkbox into a dropdown and added a test for build results. 
screen shot 2015-11-18 at 10 55 17 1
screen shot 2015-11-18 at 10 55 28

but now one of tests failed on [jenkins](https://jenkins.ci.cloudbees.com/job/plugins/job/cucumber-testresult-plugin/77/testReport/) failed which is strage for me because it is passing when i run it. does it run differently when build on server?

File: src/main/java/org/jenkinsci/plugins/cucumber/jsontestsupport/CucumberTestResultArchiver.java [hunk 1]
```diff
@@ -166,10 +179,15 @@ public boolean getIgnoreBadSteps(){
 		build.getActions().add(action);
 		CHECKPOINT.report();
 
-		if (action.getResult().getTotalCount() == action.getResult().getFailCount()){
+		if (setResult.equals("failIfAll") && action.getResult().getTotalCount() == action.getResult().getFailCount()){
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/plugins/cucumber/jsontestsupport/CucumberTestResultArchiver.java]: setResult may be `null`

**Comment by jtnord** [comment]
the failure is as the `setResult` may be `null`.

I don't think these should be strings and treated like this but Objects that can have the logic inside them (which makes for easier testing and extensibility) and internationlization of the UI.  This may require a new ExtensionPoint to be implemented, or an enum (but enums are harder to evolve/extend)

**Comment by tommywo** [comment]
Closing for now. I will submit a new pull request, if I have time to work on it.

**Pull request closed without merging**